File: solutions/clojure/acronym/3/src/acronym.clj
(ns acronym
   (:require [clojure.string :as str]))

(defn only-letters [s]
 (apply str (filter #(Character/isLetter %) s))
  )

(defn uppercase-first [s]
  (apply str (concat (str/upper-case (first s)) (rest s))))

(defn process [s]
  (if (every? #(Character/isUpperCase %) s) (first s) 
    (->> (uppercase-first s)
         (seq) 
         (filter #(Character/isUpperCase %))
         (apply str)))) 
     

(defn acronym  [phrase]
  (if (empty? phrase) "" 
   (->> (str/split (str/replace phrase "-" " " )  #"\s+")
        (map only-letters)
        (map process)
        (apply str)))
  )

File: solutions/java/acronym/2/src/main/java/Acronym.java
class Acronym {

    String acronym;

    Acronym(String phrase) {
        String acro="";
        String[] words =  phrase.split("[\s_-]+");
        for(String w: words){
            acro += w.charAt(0);
        }        
        acronym = acro.toUpperCase();
    }

    String get() {
        return acronym;
    }

}
